### ISSUE TITLE
add/insert new const/var

### ISSUE THREAD

**Issue by gp187** [dsherret/ts-simple-ast#49]
Is there a function like `addEnum` for variables and constants? 
If not, I want investigate and make it, but asking before

**Comment by dsherret** [comment]
Hey @gp187, I'm currently working on rewriting the foundation for manipulation so everything is simple (my initial pass on the implementation meant a lot of repeated code because I was figuring everything out). I skipped out on doing it for variable statements for now because I wanted to lay more of the foundation down for how manipulation of statements should work (ex. reordering statements, inserting, adding, setting a function body's text, etc).

If you would like to take a look, this would be added to `StatementedNode`. You'd need to add some `insertVariableStatement`, etc. methods and also create some "Structure" interfaces that match `VariableStatement`, `VariableDeclarationList` and `VariableDeclaration`. It's a little bit complicated though because I use code generation for some of this in order to ensure correctness (see `npm run code-generate` and `npm run code-verification` commands). I'm planning on writing up some documentation on all this in the future so anyone can come along and help maintain it.

**Comment by gp187** [comment]
I forked the repo to work on it (can't branch, no access)
Click [here](https://github.com/gp187/ts-simple-ast/commits/master) and check the last 3 commits. Some of your logic is simple, other hard to understand 

Check this [comment](https://github.com/gp187/ts-simple-ast/commit/6ce55a7c2f855109a8d92bff520b9c17667ad975) and see if you can help out. 

I'd be glad to work on this repo more as I plan to use it as well

**Comment by dsherret** [comment]
Just looking now. Yeah, this part of the code is hard to explain. I just spent some time looking at how this code be improved and I think I figured out a way to make this work without having to use code generation. It will also help implement #46.

I'm going to start work on that now and I'll get back to you when it's done.

By the way, the structures should probably almost match the nodes... so VariableStatementStructure, VariableDeclarationListStructure, and VariableDeclarationStructure.

**Comment by dsherret** [comment]
@gp187 if you merge in the latest code it should be easier to understand.

**Comment by gp187** [comment]
Great @dsherret !

**Comment by chrisgervang** [comment]
@gp187, did you have any luck making it work?

**Comment by dsherret** [comment]
This is implemented in 0.77.0.

You can add or insert variable statements to a source file, namespace, or function like declarations by calling `addVariableStatement()`, `addVariableStatements()`, `insertVariableStatement()`, or `insertVariableStatements()`.

```typescript
import Ast, {VariableDeclarationType} from "ts-simple-ast";

const variableStatement = sourceFile.addVariableStatement({
    declarationType: VariableDeclarationType.Const // defaults to "let"
    declarations: [{
        name: "myNumber",
        initializer: "5"
    }, {
        name: "myString",
        type: "string",
        initializer: `'my string'`
    }]
});
```
